PlayerControls/sources/DefaultControlsViewController.swift

```diff
@@ -132,7 +141,11 @@ public final class DefaultControlsViewController: ContentControlsViewController
             afterFadeAnimationActions.append(block)
         }
 
-        if uiProps.animationsEnabled {
+        var controlsAnimationEnabled: Bool {
+            return uiProps.animationsEnabled && hasAppeared && !isTransitioning
```

Comment:
Maybe better to move `hasAppeared` and `isTransitioning` inside some structure and on `didSet` call `setNeedsLayout()`

Reply:
You mean a struct with 2 fields and each one rewrites controlsAnimationEnabled and triggers `setNeedsLayout` on didSet?

Reply:
Not sure that I understand
> each one rewrites controlsAnimationEnabled

Reply:
@AndriiMoskvin I guess I didn't understand either.